src/compiler/codegen_c_sort.c

```diff
@@ -87,12 +87,28 @@ int gen_sort(fb_output_t *out)
         "  __flatbuffers_uoffset_write_to_pe(vec + b, tb__tmp); }\n",
         out->nsc, out->nsc);
     fprintf(out->fp,
-        "#define __%sdefine_sort_by_scalar_field(N, NK, TK, TE)\\\n"
-        "  __%sdefine_sort_by_field(N, NK, TK, TE, __%sscalar_diff, __%sscalar_swap)\n",
+            "#define __%sscalar_swap(vec, a, b, TE) __%svalue_swap(vec, a, b, TE)\n",
+        out->nsc, out->nsc);
+    fprintf(out->fp,
+            "#define __%sstring_swap(vec, a, b, TE) __%suoffset_swap(vec, a, b, TE)\n",
+        out->nsc, out->nsc);
+    fprintf(out->fp,
+            "#define __%sstruct_swap(vec, a, b, TE) __%svalue_swap(vec, a, b, TE)\n",
+        out->nsc, out->nsc);
+    fprintf(out->fp,
+            "#define __%stable_swap(vec, a, b, TE) __%suoffset_swap(vec, a, b, TE)\n",
+        out->nsc, out->nsc);
+    fprintf(out->fp,
+        "#define __%sdefine_struct_sort_by_scalar_field(N, NK, TK, TE)\\\n"
+        "  __%sdefine_sort_by_field(N, NK, TK, TE, __%sscalar_diff, __%sstruct_swap)\n",
         out->nsc, out->nsc, out->nsc, out->nsc);
+    fprintf(out->fp,
+        "#define __%sdefine_table_sort_by_scalar_field(N, NK, TK)\\\n"
+        "  __%sdefine_sort_by_field(N, NK, TK, %suoffset_t, __%sscalar_diff, __%stable_swap)\n",
+        out->nsc, out->nsc, out->nsc, out->nsc, out->nsc);
     fprintf(out->fp,
         "#define __%sdefine_sort_by_string_field(N, NK)\\\n"
-        "  __%sdefine_sort_by_field(N, NK, %sstring_t, %suoffset_t, __%sstring_diff, __%sstring_swap)\n",
+        "  __%sdefine_sort_by_field(N, NK, %sstring_t, %suoffset_t, __%sstring_diff, __%suoffset_swap)\n",
         out->nsc, out->nsc, out->nsc, out->nsc, out->nsc, out->nsc);
```

Comment:
this should have been __%stable_swap, but it amounts to the same thing.

Reply:
Passes the test as I'm sure you already know. I've also been trying it in my project and it's working fine so far there as well. :+1: